processor.py

```diff
@@ -66,8 +66,7 @@ def __init__(self, **args):
     def get_parameters(self):
         """returns a dictionary with the processor's stored parameters"""
         parameter_names = self.PARAMETERS.keys()
-        # TODO: Unresolved reference for processor
```

Comment:
I believe this code in master is faulty, but this change doesn't improve things. Please keep this PR focused on a single purpose (python3)